### PR TITLE
Add German accent

### PR DESCRIPTION
**Pull request by Psychpsyo**
## About the PR
This adds a German accent.
It does not add the lederhosen to go along with it, as I am too lazy to sprite that right now.
(I'll probably do it later though)

## Why / Balance
Cause it's funny

## Technical details
New accent

## Media
image

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes
no

**Changelog**

:cl:
- add: You can now be German on ze space station! (added German accent)

### CODE REVIEW

**Comment by Psychpsyo** [comment]
image
Uhhh... works on my machine? (investigating this now)

File: Content.Server/Speech/EntitySystems/GermanAccentSystem.cs
Comment (lzk228) [inline Content.Server/Speech/EntitySystems/GermanAccentSystem.cs:62]: bruh moment tbh

File: Resources/Locale/en-US/accent/german.ftl
Comment (lzk228) [inline Resources/Locale/en-US/accent/german.ftl:121]: uppercase

File: Resources/Locale/en-US/accent/german.ftl
Comment (slarticodefast) [reply Resources/Locale/en-US/accent/german.ftl:125]: wonderful should stay wunderbar
fantastisch for fantastic is good

**Comment by slarticodefast** [comment]
Should be
sec -> Polizei
similar to the italian accent

**Comment by superjj18** [comment]
a -> ein perhaps?

**Comment by lzk228** [comment]
> a -> ein perhaps?

since it depends on gender, better to not

**Comment by Psychpsyo** [comment]
> Should be sec -> Polizei similar to the italian accent

Was brought up in Discord already and I feel like it'd be kinda meh:
image

File: Resources/Prototypes/Traits/speech.yml
Comment (Psychpsyo) [reply Resources/Prototypes/Traits/speech.yml]: Done, for the entire file.

File: Resources/Locale/en-US/accent/german.ftl
Comment (Psychpsyo) [reply Resources/Locale/en-US/accent/german.ftl]: Done.

File: Resources/Locale/en-US/accent/german.ftl
Comment (Psychpsyo) [reply Resources/Locale/en-US/accent/german.ftl]: Good one, done.

File: Content.Server/Speech/EntitySystems/GermanAccentSystem.cs
Comment (Psychpsyo) [reply Content.Server/Speech/EntitySystems/GermanAccentSystem.cs:62]: bruh moment indeed

**Comment by timurjavid** [comment]
woman/lady -> fräulein?
man -> mann?

File: Resources/Locale/en-US/accent/german.ftl
Comment (lzk228) [inline Resources/Locale/en-US/accent/german.ftl:9]: ze sounds like "ce" (c like in center) do i suggest to change this replacement to `de`

Comment (Psychpsyo) [reply Resources/Locale/en-US/accent/german.ftl:9]: It's not exactly like in center, but a main part of a strong German accent is the 'th' turning into more of an s/z sound cause 'th' isn't a thing in German and 's' is the closest to it.

The french accent already does this (though it does ``the`` => ``'ze``) and this is pretty much the same thing.

This'd also need to replace "that", "this" and so on with "dat" and "dis" which, to me, reads as internet silly speak.

Comment (timurjavid) [reply Resources/Locale/en-US/accent/german.ftl:9]: It's an accent, `ze` isn't meant to be read in German but in English. It's how a typical German accent sounds like when they try to pronounce `the`. Ex: see [TF2 Medic](https://wiki.teamfortress.com/wiki/Medic_voice_commands#Voice_Menu_2)

Comment (Psychpsyo) [reply Resources/Locale/en-US/accent/german.ftl:9]: > also `the` is `der/die/das` in german so its really better

It already turns like 30% of all ``the`` into ``das``, before the replacements.
I think doing more would get a bit much and I think doing only ``das`` just makes it funnier since the gender won't match anyways and ``das`` feels like *the* German article. (also "die" might lead to odd sentenced)

**Comment by Psychpsyo** [comment]
> woman/lady -> fräulein? man -> mann?

Also did the plurals and gentleman/gentlemen -> herr/herren

**Comment by KingFroozy** [comment]
yes. please. also, idk if it's too much but it would be funny if worst became wurst

**Comment by Psychpsyo** [comment]
> also, idk if it's too much but it would be funny if worst became wurst

Not sure if it'd be too much but it just feels kinda... nonsensical? Like, that's two completely different things that are just spelled similarly.
I also don't think an accent should auto-insert puns like that, especially on common words.
(+sausages already get turned into Wieners and Frankfurters)

**Comment by Everturning** [comment]
Plans to make W -> V? Please?

**Comment by Psychpsyo** [comment]
> Plans to make W -> V? Please?

Thought about it, but I don't think it'd read particularly well or German.
Germans don't pronounce "lower" like "lover" or "glow" like "glove".
Also things like "pavk" and "havk" just don't make sense.

**Comment by luckyshotpictures** [comment]
This is honestly fine without the random ö/ü/ä shit. (e.g. Sud and Süd are two different things) I still don't like it, but meh. Whatever.

**Review by slarticodefast — CHANGES_REQUESTED** [review]
Works great. Found some minor code convention and IDE warning nitpicks

**Comment by Psychpsyo** [comment]
> Works great. Found some minor code convention and IDE warning nitpicks

All taken care of.

File: Resources/Locale/en-US/accent/german.ftl
Comment (UbaserB) [inline Resources/Locale/en-US/accent/german.ftl:27]: remove this translation

File: Resources/Locale/en-US/accent/german.ftl
Comment (UbaserB) [inline Resources/Locale/en-US/accent/german.ftl:49]: remove this translation

File: Resources/Locale/en-US/accent/german.ftl
Comment (UbaserB) [inline Resources/Locale/en-US/accent/german.ftl:52]: remove this translation

File: Resources/Locale/en-US/accent/german.ftl
Comment (UbaserB) [inline Resources/Locale/en-US/accent/german.ftl:55]: remove this translation

File: Resources/Locale/en-US/accent/german.ftl
Comment (UbaserB) [inline Resources/Locale/en-US/accent/german.ftl:64]: an alternative is "pass auf"

File: Resources/Locale/en-US/accent/german.ftl
Comment (UbaserB) [inline Resources/Locale/en-US/accent/german.ftl:82]: optionally could be "danke dir"

File: Resources/Locale/en-US/accent/german.ftl
Comment (UbaserB) [inline Resources/Locale/en-US/accent/german.ftl:94]: you could add "geist" or "gespenst" before poltergeist

File: Resources/Locale/en-US/accent/german.ftl
Comment (UbaserB) [inline Resources/Locale/en-US/accent/german.ftl:109]: remove this translation

File: Resources/Locale/en-US/accent/german.ftl
Comment (UbaserB) [inline Resources/Locale/en-US/accent/german.ftl:112]: 37 and 38 could optionally be replaced with "arzt" (pl. "ärzte")

File: Resources/Locale/en-US/accent/german.ftl
Comment (UbaserB) [inline Resources/Locale/en-US/accent/german.ftl:118]: alternatively could be "blöd" or "doof"

File: Resources/Locale/en-US/accent/german.ftl
Comment (UbaserB) [inline Resources/Locale/en-US/accent/german.ftl:136]: remove this translation

File: Resources/Locale/en-US/accent/german.ftl
Comment (UbaserB) [inline Resources/Locale/en-US/accent/german.ftl:142]: remove this translation

File: Resources/Locale/en-US/accent/german.ftl
Comment (UbaserB) [inline Resources/Locale/en-US/accent/german.ftl:145]: remove this translation

File: Resources/Locale/en-US/accent/german.ftl
Comment (UbaserB) [inline Resources/Locale/en-US/accent/german.ftl:151]: krankenwagen -> rettungswagen

File: Resources/Locale/en-US/accent/german.ftl
Comment (UbaserB) [inline Resources/Locale/en-US/accent/german.ftl:155]: the translation for farewell in specific should be "lebewohl"

File: Resources/Locale/en-US/accent/german.ftl
Comment (UbaserB) [inline Resources/Locale/en-US/accent/german.ftl:181]: remove this translation

File: Resources/Locale/en-US/accent/german.ftl
Comment (UbaserB) [inline Resources/Locale/en-US/accent/german.ftl:211]: "frau" and "frauen" here

File: Content.Server/Speech/EntitySystems/GermanAccentSystem.cs
Comment (UbaserB) [inline Content.Server/Speech/EntitySystems/GermanAccentSystem.cs:65]: this whole thing where the AOU becomes ÄÖÜ should be removed altogether

File: Resources/Locale/en-US/accent/german.ftl
Comment (UbaserB) [inline Resources/Locale/en-US/accent/german.ftl:21]: wiener -> würstchen

File: Resources/Locale/en-US/accent/german.ftl
Comment (UbaserB) [inline Resources/Locale/en-US/accent/german.ftl:12]: scheisse -> krankenhaus

File: Resources/Locale/en-US/accent/german.ftl
Comment (UbaserB) [inline Resources/Locale/en-US/accent/german.ftl:34]: remove this translation

**Review by UbaserB — CHANGES_REQUESTED** [review]
I am not German myself, but several Germans in the staff team have reviewed this PR and I have summarised what needs to be done.

File: Resources/Locale/en-US/accent/german.ftl
Comment (slarticodefast) [reply Resources/Locale/en-US/accent/german.ftl:64]: Achtung is better in my opinion.

File: Resources/Locale/en-US/accent/german.ftl
Comment (slarticodefast) [reply Resources/Locale/en-US/accent/german.ftl:52]: I think this one is perfectly fine to keep.

File: Resources/Locale/en-US/accent/german.ftl
Comment (UbaserB) [reply Resources/Locale/en-US/accent/german.ftl:12]: sorry about that

File: Resources/Locale/en-US/accent/german.ftl
Comment (slarticodefast) [reply Resources/Locale/en-US/accent/german.ftl:145]: Nah, keep this one. It is perfectly fine.

File: Resources/Locale/en-US/accent/german.ftl
Comment (slarticodefast) [reply Resources/Locale/en-US/accent/german.ftl:94]: Poltergeist is better in my opinion since it is also used in english, whereas Geist or Gespenst would be less known. Also an old horror movie title

File: Resources/Locale/en-US/accent/german.ftl
Comment (slarticodefast) [reply Resources/Locale/en-US/accent/german.ftl:151]: 9783866062047-us

There is a very technical difference that people working with one can tell, but Krankenwagen is colloquially fine.

File: Resources/Locale/en-US/accent/german.ftl
Comment (slarticodefast) [reply Resources/Locale/en-US/accent/german.ftl:82]: I prefer Dankeschön. Might be a dialect thing.

File: Resources/Locale/en-US/accent/german.ftl
Comment (slarticodefast) [reply Resources/Locale/en-US/accent/german.ftl:12]: Nothing to be sorry about, other languages are complicated :)

File: Resources/Locale/en-US/accent/german.ftl
Comment (Psychpsyo) [reply Resources/Locale/en-US/accent/german.ftl:82]: Also, "danke dir" would not work for "thank you" if it's addressing multiple people.

File: Resources/Locale/en-US/accent/german.ftl
Comment (Psychpsyo) [reply Resources/Locale/en-US/accent/german.ftl:112]: I don't think any English speaker would know what that means though.

File: Resources/Locale/en-US/accent/german.ftl
Comment (Psychpsyo) [reply Resources/Locale/en-US/accent/german.ftl:118]: English speakers don't really know those though, do they?
"dumm", in my opinion, works because it's close do "dumb".

File: Resources/Locale/en-US/accent/german.ftl
Comment (Psychpsyo) [reply Resources/Locale/en-US/accent/german.ftl:151]: This one's here for the meme, if it were "Rettungswagen", it might as well get removed since most English speakers have probably never heard that word.

File: Resources/Locale/en-US/accent/german.ftl
Comment (Psychpsyo) [reply Resources/Locale/en-US/accent/german.ftl:27]: The singular one as well or only the plural one?

File: Resources/Locale/en-US/accent/german.ftl
Comment (UbaserB) [reply Resources/Locale/en-US/accent/german.ftl:118]: I'm a native english speaker and doof is definitely easily translatable

File: Resources/Locale/en-US/accent/german.ftl
Comment (UbaserB) [reply Resources/Locale/en-US/accent/german.ftl:27]: both

File: Resources/Locale/en-US/accent/german.ftl
Comment (UbaserB) [reply Resources/Locale/en-US/accent/german.ftl:112]: that's fair

**Comment by Psychpsyo** [comment]
@UbaserB Thanks for all the feedback, I would just like to get a clarification on a thing that kinda permeates a lot of these:
For many of them, I, as a German myself, cannot tell why they should be removed. You also keep calling them "translations".

In my understanding until now, accent word replacement aren't meant to be actual, accurate translations but rather replacements that add foreign flavour and some humor to an accent while still keeping it legible to English speakers.
So I was wondering: Is that not the case or is there some general philosophy to picking these that I am unaware of?

I'm willing to (in some places begrudgingly) make these changes, I'd just like to understand what is wrong with the current words so that I can avoid those mistakes in the future.

**Comment by UbaserB** [comment]
> @UbaserB Thanks for all the feedback, I would just like to get a clarification on a thing that kinda permeates a lot of these: For many of them, I, as a German myself, cannot tell why they should be removed. You also keep calling them "translations".
> 
> In my understanding until now, accent word replacement aren't meant to be actual, accurate translations but rather replacements that add foreign flavour and some humor to an accent while still keeping it legible to English speakers. So I was wondering: Is that not the case or is there some general philosophy to picking these that I am unaware of?
> 
> I'm willing to (in some places begrudgingly) make these changes, I'd just like to understand what is wrong with the current words so that I can avoid those mistakes in the future.

From what I understand of the discussions thus far in the maintainer team, there is preference of realism over humor. Keeping it "understandable" is an important aspect for sure, but it is preferred for the translations to be commonly-used words (or accurate).

File: Resources/Locale/en-US/accent/german.ftl
Comment (Psychpsyo) [reply Resources/Locale/en-US/accent/german.ftl:151]: Also, "Krankenwagen" and "Rettungswagen" are both accurate translations.
I had to look this up just now, the exact distinction is that a Krankenwagen is mainly just to get an injured person to the hospital safely, while a Rettungswagen is for more severe cases like strokes or heart attacks where more extensive medical supplies are needed during the transport. (But even a Krankenwagen comes with all the essentials like bandages, oxygen, a defibrillator...)

TL;DR:
Krankenwagen seems perfectly fine unless I misunderstand what an ambulance is.

File: Resources/Locale/en-US/accent/german.ftl
Comment (Psychpsyo) [reply Resources/Locale/en-US/accent/german.ftl:118]: Fair.
Since there is currently no system to have multiple replacements for one word, I'm changing it to
dumb => dumm
stupid => doof

File: Resources/Locale/en-US/accent/german.ftl
Comment (Psychpsyo) [reply Resources/Locale/en-US/accent/german.ftl:136]: Does this also apply to "commander"?
Since that's also just the German form of the word and I can't tell if this is because this replacement in particular is inaccurate somehow or if I'm supposed to remove all military ranks.

File: Resources/Locale/en-US/accent/german.ftl
Comment (Psychpsyo) [reply Resources/Locale/en-US/accent/german.ftl:21]: That's a mouthful to read in English, I'd rather remove the plural alltogether.

File: Resources/Locale/en-US/accent/german.ftl
Comment (Psychpsyo) [reply Resources/Locale/en-US/accent/german.ftl:160]: Resolved by just removing "farewell"

File: Resources/Locale/en-US/accent/german.ftl
Comment (Psychpsyo) [reply Resources/Locale/en-US/accent/german.ftl:95]: There's not a way to add multiple replacements for the same word and I didn't put those because they aren't really well known to non-german speakers. (in my understanding)

**Comment by Psychpsyo** [comment]
The remaining comments have been discussed on Discord at https://discord.com/channels/310555209753690112/1272915820233625672 and the relevant changes have been made.

**Review by UbaserB — APPROVED** [review]

**Comment by Everturning** [comment]
> > Plans to make W -> V? Please?
> 
> I had already thought about it, but I don't think it reads particularly well or German. Germans don't pronounce "lower" like "lover" or "glow" like "glove". Also things like "pavn" and "havk" just don't work.

Just make it for the beginning of vords and then for things like "one" turn it into "vun"

**Comment by luckyshotpictures** [comment]
> > > Plans to make W -> V? Please?
> > 
> > 
> > I had already thought about it, but I don't think it reads particularly well or German. Germans don't pronounce "lower" like "lover" or "glow" like "glove". Also things like "pavn" and "havk" just don't work.
> 
> Just make it for the beginning of vords and then for things like "one" turn it into "vun"

This doesn't really make any sense.

**Comment by Psychpsyo** [comment]
Also, the W -> V thing would need to happen at the same time as the word replacements since otherwise, either "warning" -> "warnung" breaks or we'd get "sausage" -> "vurst".
So it requires extending replacement accents with regex support or similar which is not a thing I am interested in lumping in with this PR.
That can happen later if someone wants to do it.

**Comment by Psychpsyo** [comment]
Also, @slarticodefast, Github is telling me there is still an outstanding requested change from you on this, but on my end it looks like all the requested changes are marked as resolved.
Is there something I'm missing or do you just need to mark them as resolved on your end or so?

**Comment by slarticodefast** [comment]
> Github is telling me there is still an outstanding requested change from you on this, but on my end it looks like all the requested changes are marked as resolved.
Is there something I'm missing or do you just need to mark them as resolved on your end or so?

That was just a reminder for myself to give it one last test with the new changes before merging. There were no new complaints from anyone, so I will merge it in a moment if everythign works.

**Review by slarticodefast — APPROVED** [review]

**Pull request merged**
